packages/migrations/src/load-migrations.ts

```diff
@@ -24,26 +24,28 @@ export interface MigrationFile {
   timestamp: string;
 }
 
+export function extractTimestampFromFilename(filename: string): string {
+  const match = filename.match(MIGRATION_FILENAME_REGEX);
+  if (!match) {
+    throw new Error(`Invalid migration filename: ${filename}`);
+  }
+  const timestamp = match.at(1) ?? null;
+  if (timestamp === null) {
+    throw new Error(`Migration ${filename} does not have a timestamp`);
+  }
+  return timestamp;
+}
+
 export async function readAndValidateMigrationsFromDirectory(
   dir: string,
   extensions: string[],
 ): Promise<MigrationFile[]> {
   const migrationFiles = (await fs.readdir(dir)).filter((m) =>
-    extensions.some((e) => m.endsWith(e)),
+    extensions.includes('.' + (m.split('.', 2).pop() ?? '')),
```

Comment:
This is very opaque and definitely needs an explanation of what's happening here.